packages/php-wasm/universal/src/lib/base-php.ts

```diff
@@ -397,7 +397,7 @@ export abstract class BasePHP implements IsomorphicLocalPHP {
 						 *
 						 * @TODO: Determine whether this is a bug in emscripten or in our code.
 						 */
-						await await this[__private__dont__use].ccall(
+						this[__private__dont__use].ccall(
```

Comment:
Why remove async/await here? See the comment in the code

Reply:
This is not needed. The reason is that the double await doesn't needed either, so you can toss that.
You can verify that this will work, by simulating that using the following simple example

```
async function test3(){
    return 4
}

async function test2(){
    return test(3)
}

function test(){
    return new Promise((res, rej) => {
        return res(
             test2() // <---- This is a promise that returns a promise. Resolve handles it as it should.No need for extra async here
            )
    })
}

test().then(result => {
    console.log(result) // <--- This prints 4, and not Promise<4>. It resolves successfully.
})
```

Reply:
@kozer asynchronous exceptions won't be caught by `try/catch` without the `await`, though

Reply:
@adamziel , I think  I need some help here.
I added the following test:
```
		it('throw an error in case of errornous scriptPath ', async () => {
			try {
				await php.run({
					scriptPath: 'index.php',
				});
			} catch (e) {
				expect(e.message).toMatch('Program terminated with exit(1)');
			}
		});
```
in `php.spec.ts`.
Ignore the `toMatch` condition for now.
I run it both against the trunk and my branch.
The test fails with this error `expected 'Aborted(). Build with -sASSERTIONS fo…' to include 'Program terminated with exit(1)'`
which is expected, but the main problem is that I get the following warning when I run the tests:
```
(node:399877) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 unhandledRejection listeners added to [process]. Use emitter.setMaxListeners() to increase limit
```
which makes me suspect that the error is coming from elsewhere.
I also noticed that `onAbort` is called, in case that helps you.

Also, by trying to write this test, I noticed that the `catch` clause is executed in both branches.
In both cases the following line throws:
```
		const { headers, httpStatusCode } = this.#getResponseHeaders();
```

Reply:
The `MaxListenersExceededWarning` is irritating, I don't know where it's coming from but it's been there since the very first test when we didn't even support Asyncify. Must be something specific to our vite setup – all Playground-related code should clean up its listeners. I just created an issue: https://github.com/WordPress/wordpress-playground/issues/350

Reply:
As for the error – Emscripten messages are pretty crude. Playground rewrites a bunch of them, but not this one apparently. Anyway, the only thing that matters is: does the entire Node.js process crash? With `await` and `try/catch` in place we should be able to handle it gracefully – that's what the entire rethrowing gymnastics in this function are for. I'm looking into this now.

Reply:
I can see a single `await` does the trick there, though, which is not what I've seen before. Hm. Now I can't even write a function that returns `Promise<Promise<any>>`, JS always flattens it to just `Promise<any>`. But I am 100% sure it actually did return a promise that resolved to another promise.

Reply:
So weird! https://promisesaplus.com/ contradicts what I'm saying. And yet, I know there was a scenario where double `await` was necessary.

Reply:
Alright, I can't make JS do it again. Let's refactor to something like below to indicate the problem existed but take the `await await` off the main code path:

```js
let response = await this{_…];

/* This is a bug found blah blab blah
 * response should never be a Promise according to https://promisesaplus.com/
 * This code shouldn't be necessary but let's keep it here for now
 * since there was an actual weird case where `await` resolved to a Promise.
 */
if (response instanceof Promise) {
    response = await response;
}
```

Reply:
Sure! Thanks for your effort!
I think the not need for that is because we always return from the function.
If you have a promise that is not returned somewhere deep into the code, this will be reproducible.
I'll make the change and push it!
What about the test? Do you think the test you proposed above, can be used?

Reply:
Oh I think that I understand what you mean by the `await` in there @adamziel . You mean that the inner `try..catch` doesn't resolve, right?
However, when I added a `console.log` in there I was able to see the error.

Reply:
@adamziel I change the code a bit, and it seems that your test passes now.
Can you give it a look?

Reply:
@kozer I don't see a way around having the async function – you'll see the test I proposed above just hangs if the `ccall` is not `await`-ed.

Reply:
Did you see my current changes?
It seems that the test you provided passed now.
I pushed the test here as well.

Reply:
@kozer the CI got derailed by a formatting issue, would you push a fix? I wish I had set up a pre-commit hook, that check gets me too.

Reply:
The async rejection part of this PR looks good to me 👍